docs/dag-run.rst

```diff
@@ -0,0 +1,193 @@
+and the next one will be created just after midnight on the morning of 2016-01-03 with an execution date of 2016-01-02.
+
+If the ``dag.catchup`` value had been True instead, the scheduler would have created a DAG Run 
+for each completed interval between 2015-12-01 and 2016-01-02 (but not yet one for 2016-01-02, 
+as that interval hasn’t completed) and the scheduler will execute them sequentially. 
+
+Catchup is also triggered when you turn off a DAG for a specified period of time and then re-enable.
+
+This behavior 
+is great for atomic datasets that can easily be split into periods. Turning catchup off is great 
+if your DAG Runs perform backfill internally.
+
+
+Backfill
+---------
+There can be the case when you may want to run the dag for a specified historical period e.g. a data pipeline
+which dumps data in a DFS every day and another pipeline which requires last 1 month of data in DFS. 
+This is known as Backfill.
+
+You may want to backfill the data even in the cases when catchup is disabled. This can be done through CLI. 
+Run the below command
+
+.. code:: bash
+
+    airflow backfill -s START_DATE -e END_DATE dag_id
+
+The above command will re-run all the instances of the dag_id for all the intervals within the start date and end date.
+
+Re-run Tasks
+------------
+It can happen, that some of the tasks can fail during the scheduled run. Once you have fixed 
+the errors after going through the logs, you can re-run the tasks by clearing it for the 
+scheduled date. Clearing a task instance doesn't delete the task instance record. 
+Instead it updates ``max_tries`` to ``0`` and set the current task instance state to be ``None``, this forces the task to re-run.
+
+Select the failed task and click on **Clear**. This will clear the status of the task from 
+failed to ``None`` and the executor will re-run it.
+
+There are multiple options you can select to re-run - 
+
+* Past - All the instances of the task in the  runs before the current DAG's execution date
+* Future -  All the instances of the task in the  runs after the current DAG's execution date
+* Upstream - The upstream tasks in the current DAG
+* Downstream - The downstream tasks in the current DAG
+* Recursive - All the tasks in the child DAGs and parent DAGs
+* Failed - Only the failed tasks in the current DAG
+
+You can also clear the task through CLI using the command:
+
+.. code:: bash
+
+    airflow tasks clear dag_id -t task_regex -s START_DATE -d END_DATE
+
+This will clear all instances of the tasks matching the regex for the dag_id which have run during 
+the mentioned interval. For more options, you can run the command:
+
+.. code:: bash
+
+    airflow tasks clear -h
+
+**Note**: When clearing a set of tasks’ state in hope of getting them to re-run, it is important 
+to keep in mind the DAG Run’s state too as it defines whether the scheduler should look
+into triggering tasks for that run.
```

Comment:
I think that if you clear tasks it will also set the dagrun state to running? But I'm not certain of that.